### PR TITLE
JSON stringify types over the ffi

### PR DESCRIPTION
**Pull request by KendallWeihe**
I JSON-stringified Offering's over the wasm FFI & ran some experiments

- Reduced bundle size by 100KB
- Increased test vector duration from 35ms to 32ms 

It is surprisingly faster! I wouldn't have guessed that.

### CODE REVIEW

File: bound/typescript/src/presentation-definition.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  from?: string;
+  from_nested?: SubmissionRequirement[];
+  max?: number;
+  min?: number;
+  name?: string;
+  purpose?: string;
+  rule: SubmissionRequirementRule;
+};
+
+// todo make enum
```
Comment (KendallWeihe) [inline bound/typescript/src/presentation-definition.ts:53]: I would want to make this and `Optionality` above actual ts enums, or type unions to be more exact

**Review by nitro-neal — APPROVED** [review]
awesome job

File: bound/typescript/src/http-client/exchanges.ts [hunk 2]
```diff
@@ -82,39 +82,48 @@ export class Exchange {
...
-        wasmExchange.order_statuses
-          ? wasmExchange.order_statuses.map(OrderStatus.fromWASM)
-          : undefined,
-        wasmExchange.close ? Close.fromWASM(wasmExchange.close) : undefined
-      );
-    } catch (error) {
-      throw tbdexError(error);
-    }
-  };
+  static fromJSONString = (json: string): Exchange => {
```
Comment (nitro-neal) [inline bound/typescript/src/http-client/exchanges.ts:85]: may be a bit cleaner something like this (from chad)
```static fromJSONString = (json: string): Exchange => {
  const obj = JSON.parse(json);

  const createInstance = (ClassType: any, objPart?: any) => 
    objPart ? new ClassType(objPart.metadata, objPart.data, objPart.signature) : undefined;

  const rfq = createInstance(Rfq, obj.rfq);
  const quote = createInstance(Quote, obj.quote);
  const order = createInstance(Order, obj.order);
  const orderInstructions = createInstance(OrderInstructions, obj.orderInstructions);
  const cancel = createInstance(Cancel, obj.cancel);
  const orderStatuses = obj.orderStatuses
    ? obj.orderStatuses.map((x: any) => createInstance(OrderStatus, x))
    : undefined;
  const close = createInstance(Close, obj.close);

  return new Exchange(rfq, quote, order, orderInstructions, cancel, orderStatuses, close);
}
```

File: bindings/tbdex_wasm/src/http_client/offerings.rs [hunk 3]
```diff
@@ -1,15 +1,20 @@
...
+pub async fn get_offerings(pfi_did_uri: &str) -> Result<String> {
+    let offerings = tbdex::http_client::offerings::get_offerings(pfi_did_uri)
         .await
-        .map_err(map_err)?
-        .into_iter()
-        .map(|offering| offering.into())
-        .collect())
+        .map_err(map_err)?;
+
+    let mut json = "[".to_string();
```
[the remark this answers is not in the document]
Comment (KendallWeihe) [reply bindings/tbdex_wasm/src/http_client/offerings.rs:11]: We need to change this actually. I was firing and forgetting, and initially thought to re-use our existing ToJson implementation per Offering. But it's better performance wise to just run a single serialization over the entire vector.

**Pull request merged**